cmd/controller/main.go

```diff
@@ -59,33 +60,36 @@ import (
 )
 
 const (
-	enableStackdriverMetricsFlag = "stackdriver-exporter"
-	stackdriverLabels            = "stackdriver-labels"
-	enablePrometheusMetricsFlag  = "prometheus-exporter"
-	projectIDFlag                = "gcp-project-id"
-	sidecarImageFlag             = "sidecar-image"
-	sidecarCPURequestFlag        = "sidecar-cpu-request"
-	sidecarCPULimitFlag          = "sidecar-cpu-limit"
-	sidecarMemoryRequestFlag     = "sidecar-memory-request"
-	sidecarMemoryLimitFlag       = "sidecar-memory-limit"
-	sdkServerAccountFlag         = "sdk-service-account"
-	pullSidecarFlag              = "always-pull-sidecar"
-	minPortFlag                  = "min-port"
-	maxPortFlag                  = "max-port"
-	additionalPortRangesFlag     = "additional-port-ranges"
-	certFileFlag                 = "cert-file"
-	keyFileFlag                  = "key-file"
-	numWorkersFlag               = "num-workers"
-	apiServerSustainedQPSFlag    = "api-server-qps"
-	apiServerBurstQPSFlag        = "api-server-qps-burst"
-	logDirFlag                   = "log-dir"
-	logLevelFlag                 = "log-level"
-	logSizeLimitMBFlag           = "log-size-limit-mb"
-	kubeconfigFlag               = "kubeconfig"
-	allocationBatchWaitTime      = "allocation-batch-wait-time"
-	defaultResync                = 30 * time.Second
-	podNamespace                 = "pod-namespace"
-	leaderElectionFlag           = "leader-election"
+	enableStackdriverMetricsFlag         = "stackdriver-exporter"
+	stackdriverLabels                    = "stackdriver-labels"
+	enablePrometheusMetricsFlag          = "prometheus-exporter"
+	projectIDFlag                        = "gcp-project-id"
+	sidecarImageFlag                     = "sidecar-image"
+	sidecarCPURequestFlag                = "sidecar-cpu-request"
+	sidecarCPULimitFlag                  = "sidecar-cpu-limit"
+	sidecarMemoryRequestFlag             = "sidecar-memory-request"
+	sidecarMemoryLimitFlag               = "sidecar-memory-limit"
+	sidecarRunAsNonRootFlag              = "sidecar-run-as-nonroot"
```

Comment:
You went all out on the flags! I don't think we need flags for `RunAsNonRoot` or `AllowPrivilegeEscalation`, those can be hardcoded. We don't want the user to change them, IMO.

Reply:
Agreed! I've removed these 2 flags.